File: Backend/Clinica.Dominio/Entidades/Turno.cs
using Clinica.Dominio.Enumeraciones;

namespace Clinica.Dominio.Entidades;

public class Turno
{
    public int Id { get; set; }

    public int PacienteId { get; set; }
    public Paciente Paciente { get; set; } = null!;

    public int ProfesionalId { get; set; }
    public Profesional Profesional { get; set; } = null!;

    public DateOnly Fecha { get; set; }
    public TimeOnly Horario { get; set; }
    public EstadoTurno Estado { get; set; } = EstadoTurno.Pendiente;

    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaModificacion { get; set; }

    // No se declara una propiedad de concurrencia explícita: se usa la columna
    // de sistema "xmin" de PostgreSQL como token de concurrencia optimista,
    // configurada como shadow property en TurnoConfiguracion (UseXminAsConcurrencyToken).
}

File: Frontend/src/modelos/turno.ts
export type EstadoTurno = 'Pendiente' | 'Confirmado' | 'Cancelado' | 'Atendido';

export const ESTADOS_TURNO: EstadoTurno[] = ['Pendiente', 'Confirmado', 'Cancelado', 'Atendido'];

export interface Turno {
  id: number;
  pacienteId: number;
  pacienteNombreCompleto: string;
  profesionalId: number;
  profesionalNombreCompleto: string;
  profesionalEspecialidad: string;
  fecha: string; // formato "AAAA-MM-DD"
  horario: string; // formato "HH:mm:ss"
  estado: EstadoTurno;
  fechaCreacion: string;
}

export interface CrearTurnoDto {
  pacienteId: number;
  profesionalId: number;
  fecha: string;
  horario: string;
}

export interface ActualizarTurnoDto extends CrearTurnoDto {
  estado: EstadoTurno;
}

export interface TurnoFiltro {
  profesionalId?: number;
  fecha?: string;
  estado?: EstadoTurno;
}
